Make the seed a global variable

davidbbeyer opened the pull request:
Solves #44, #43 

@pm-blanco I have made the `SEED` a mandatory attribute of the `pmb` class, which is now used to set up the various reaction methods. Furthermore, I have added an attribute `rng` that is initialized using the provided `SEED` and used in `generate_random_points_in_a_sphere`. This way it is avoided that the `rng`, which was before initialized within `generate_random_points_in_a_sphere`, is each time re-initialized with the same seed. 

All samples, tests, etc. have been updated accordingly.

Moreover, I have also fixed the bug discovered by @kosovan regarding non-cubic simulation boxes (#43)

pm-blanco commented:
thanks @davidbbeyer for taking care of this, we should make a couple of CI tests for this:
- one that tests that `center_molecule_in_simulation_box` places the molecule in the center of the simulation box, for both cubic and non cubic simulation boxes. Probably could be part of `testsuite/create_molecule_position_test.py`
- We should have a CI test that checks that pyMBE produces deterministic results for a given `SEED`.  I am not completely sure what is the best way we could check this. A simple test could be to create a somewhat complex molecule twice into espresso with two different instances of pyMBE with the same `SEED` and check that one gets the same structure

davidbbeyer commented:
@pm-blanco I have implemented your suggestions:

- `testsuite/create_molecule_position_test.py` now contains two tests that check if the function `center_molecule_in_simulation_box` works correctly. Note that I had to slightly modify the function `center_molecule_in_simulation_box`, since it was performing a check that was not strictly necessary and in some case might cause that the molecule is _not_ centered in the box (depending on the order that objects with a given molecule id are added to the dataframe). I also adjusted the function `create_molecule` such that the returned dictionary now actually matches the description in the docstring.

- I added a test `testsuite/seed_test.py` that checks if a molecule is created at exactly the same position if the same global seed is provided to pyMBE.

pm-blanco commented on pyMBE.py:
```diff
@@ -450,15 +455,14 @@ def center_molecule_in_simulation_box(self, molecule_id, espresso_system):
-        box_center = [espresso_system.box_l[0]/2.0]*3
-        pmb_type = self.df.loc[self.df['molecule_id']==molecule_id].pmb_type.values[0]
-        pmb_objects = ['protein','molecule','peptide']
-        if pmb_type in pmb_objects:
```
You are right that this check is not really necessary, However, what is missing is a sanity check to ensure that the input molecule_id is defined in the pmb.df, otherwise the user will get a somewhat confusing pandas error

davidbbeyer replied:
I have added such a check.

pm-blanco reviewed (changes requested):
Apart from this minor things, it looks very good to me. Thank you for taking care of the bug in the return info in `create_molecule`

davidbbeyer commented on testsuite/seed_test.py:
```diff
@@ -0,0 +1,61 @@
+
+espresso_system = espressomd.System(box_l = [100]*3)
+
+def run_peptide_simulation(SEED):
```
done

pm-blanco reviewed (approved):
it looks good to me
